File: Entities/Reserva.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace ProyectoFinal.Entities
{
    public class Reserva
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string NameUsuario { get; set; }
        public string? FechaInicio { get; set; }
        public string? FechaFin { get; set; }

        [ForeignKey("ViviendaId")]
        public Vivienda? Vivienda { get; set; }

        public int ViviendaId { get; set; }

        public Reserva(string nameUsuario)
        {
            NameUsuario = nameUsuario;
        }
    }
}

File: ClientApp/src/app/shared/reserva.ts
export interface IReserva {
  id: number;
  fechaInicio: string;
  fechaFin: string;
  nameUsuario: string;
}

export interface IReservaCreate {
  fechaInicio: string;
  fechaFin: string;
  nameUsuario: string;
}
